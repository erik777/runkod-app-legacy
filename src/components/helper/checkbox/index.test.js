import React from 'react';
import Checkbox from './index';
import TestRenderer from 'react-test-renderer';


test('1- default', () => {
  const props = {};

  const renderer = TestRenderer.create(
    <Checkbox {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});

test('2- checked', () => {
  const props = {
    checked: true
  };

  const renderer = TestRenderer.create(
    <Checkbox {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});


test('3- disabled', () => {
  const props = {
    disabled: true
  };

  const renderer = TestRenderer.create(
    <Checkbox {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});

test('4- checked + disabled', () => {
  const props = {
    checked: true,
    disabled: true
  };

  const renderer = TestRenderer.create(
    <Checkbox {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});

