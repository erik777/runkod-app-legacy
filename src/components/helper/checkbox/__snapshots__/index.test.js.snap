// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default 1`] = `
<div
  className="checkbox_"
  onClick={[Function]}
/>
`;

exports[`2- checked 1`] = `
<div
  className="checkbox_ checked"
  onClick={[Function]}
>
  <svg
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
    />
  </svg>
</div>
`;

exports[`3- disabled 1`] = `
<div
  className="checkbox_ disabled"
  onClick={[Function]}
/>
`;

exports[`4- checked + disabled 1`] = `
<div
  className="checkbox_ checked disabled"
  onClick={[Function]}
>
  <svg
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
    />
  </svg>
</div>
`;
